Source/NameServer/AddressBook.cpp

```diff
@@ -0,0 +1,1361 @@
+    if(TRec == PNULL)
+    {
+        //throw("Task Does Not Exist");
+        return ERR_TASK_NOT_FOUND;
+    }
+
+    TRec->State = State;
+
+    return SUCCESS;
+}
+
+/*==============================================================================
+ * TaskExecPath:    Get the path to the executables for the task
+ *============================================================================*/
+std::string AddressBook::TaskExecPath(std::string &TaskName)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        //throw("Task Does Not Exist");
+        return "";
+    }
+
+    return TRec->ExecPath;
+}
+
+/*==============================================================================
+ * TaskExecPath:    Set the path to the executables for the task
+ *============================================================================*/
+unsigned AddressBook::TaskExecPath(std::string &TaskName, std::string &NewPath)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        ERETURN("Task Does Not Exist", ERR_TASK_NOT_FOUND);
+    }
+
+    TRec->ExecPath = NewPath;
+    return SUCCESS;
+}
+
+/*==============================================================================
+ * TaskValid:   Indicates whether the Task's data is valid
+ *============================================================================*/
+bool AddressBook::TaskValid(std::string &TaskName)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        ERETURN("Task Does Not Exist", false);
+    }
+
+    return TRec->TaskValid;
+}
+
+/*==============================================================================
+ * TaskMapValid:    Indicates whether the Task's mappings are valid
+ *============================================================================*/
+bool AddressBook::TaskMapValid(std::string &TaskName)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        ERETURN("Task Does Not Exist", false);
+    }
+
+    return TRec->MapValid;
+}
+
+/*==============================================================================
+ * TaskLinkValid:   Indicates whether the Task's linkage is valid
+ *============================================================================*/
+bool AddressBook::TaskLinkValid(std::string &TaskName)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        ERETURN("Task Does Not Exist", false);
+    }
+
+    return TRec->LinkValid;
+}
+
+
+/*==============================================================================
+ * RebuildTask: Rebuild the Task's MessageType/DeviceType maps
+ *============================================================================*/
+unsigned AddressBook::RebuildTask(std::string &TaskName)
+{
+    TaskRecord_t *TRec = FindTask(TaskName);
+    if(TRec == PNULL)
+    {
+        ERETURN("Task Does Not Exist", ERR_TASK_NOT_FOUND);
+    }
+
+    //TODO: test this
```

Comment:
Ominous!

Reply:
That's a straggler - the rebuild code has been tested using a selection of test-cases before @AlexRast started the integration.

From talks with ADB and @AlexRast, the rebuild functionality is hopefully going to be unused.